app/views/settings/_advanced_messenger_settings.html.erb

```diff
@@ -13,4 +13,16 @@
     <%= hidden_field_tag 'settings[disable_fix_for_scroll_to_anchor]', 0, :id => nil %>
     <%= check_box_tag 'settings[disable_fix_for_scroll_to_anchor]', 1, Setting.plugin_redmine_advanced_messenger[:disable_fix_for_scroll_to_anchor] == '1' %>
     <em class="info"><%=raw l(:disable_fix_for_scroll_to_anchor_note) %></em>
+</p>
+<p>
+    <label for="settings_select_roles_to_exclude"><%= l(:select_roles_to_exclude) %></label>
+    <% allowed_roles = Setting.plugin_redmine_advanced_messenger[:allowed_roles] || [] %>

```

Comment:
Attention: This is the list of roles that will not receive mails. So probably `roles_not_sent_email` or a better name.

Also verify that the stop email logic does indeed stop emails for the roles (and not sends email only for this roles)